classes/migration/upgrade/v3_4_0/I7725_DecisionConstantsUpdate.php

```diff
@@ -0,0 +1,207 @@
+            'updated_value' => 15,
+        ],
+
+        // \PKP\decision\Decision::REVERT_INITIAL_DECLINE
+        [
+            'stage_id' => [WORKFLOW_STAGE_ID_SUBMISSION],
+            'current_value' => 18,
+            'updated_value' => 16
+        ],
+
+        // \PKP\decision\Decision::SKIP_EXTERNAL_REVIEW
+        [
+            'stage_id' => [WORKFLOW_STAGE_ID_EDITING],
+            'current_value' => 19,
+            'updated_value' => 17,
+        ],
+
+        // \PKP\decision\Decision::BACK_FROM_PRODUCTION
+        [
+            'stage_id' => [WORKFLOW_STAGE_ID_EDITING],
+            'current_value' => 31,
+            'updated_value' => 29,
+        ],
+
+        // \PKP\decision\Decision::BACK_FROM_COPYEDITING
+        [
+            'stage_id' => [WORKFLOW_STAGE_ID_SUBMISSION, WORKFLOW_STAGE_ID_EXTERNAL_REVIEW],
+            'current_value' => 32,
+            'updated_value' => 30,
+        ],
+
+        // \PKP\decision\Decision::CANCEL_REVIEW_ROUND
+        [
+            'stage_id' => [WORKFLOW_STAGE_ID_SUBMISSION, WORKFLOW_STAGE_ID_EXTERNAL_REVIEW],
+            'current_value' => 33,
+            'updated_value' => 31,
+        ],
+    ];
+
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        try {
+            DB::beginTransaction();
+
+            collect($this->decisionMappings)
+                ->each(
+                    fn ($decisionMapping) => DB::table('edit_decisions')
+                        ->when(
+                            isset($decisionMapping['stage_id']) && !empty($decisionMapping['stage_id']),
+                            fn ($query) => $query->whereIn('stage_id', $decisionMapping['stage_id'])
+                        )
+                        ->where('decision', $decisionMapping['current_value'])
+                        ->update([
+                            'decision' => $decisionMapping['updated_value'],
+                        ])
+                );
+
+            DB::commit();
+        } catch (Throwable $exception) {
+            DB::rollBack();
+
+            error_log($exception->__toString());
+        }
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        try {
+            DB::beginTransaction();
+
+            collect($this->decisionMappings)
+                ->each(
+                    fn ($decisionMapping) => DB::table('edit_decisions')
+                        ->when(
+                            isset($decisionMapping['stage_id']) && !empty($decisionMapping['stage_id']),
+                            fn ($query) => $query->whereIn('stage_id', $decisionMapping['stage_id'])
+                        )
+                        ->where('decision', $decisionMapping['updated_value'])
+                        ->update([
+                            'decision' => $decisionMapping['current_value'],
+                        ])
+                );
+
+            DB::commit();
```

Comment:
> is this what transactions are meant to avoid?

No , that is not the purpose of transaction here . Basically we want to have all updated or if failed , no changes to be committed to DB so that we don't end up with unsynced values . 

@NateWr you have raised a very important point . The example you have provided , for that case , it will not cause that issue as we are using here combination of current `decision` value and `stage_id`. As those are different , it will not be an issue . 

But for the following one case

```php
// \PKP\decision\Decision::PENDING_REVISIONS
[
    'stage_id' => [WORKFLOW_STAGE_ID_EXTERNAL_REVIEW],
    'current_value' => 2,
    'updated_value' => 4,
],

// \PKP\decision\Decision::RESUBMIT
[
    'stage_id' => [WORKFLOW_STAGE_ID_EXTERNAL_REVIEW],
    'current_value' => 3,
    'updated_value' => 5,
],

// \PKP\decision\Decision::DECLINE
[
    'stage_id' => [WORKFLOW_STAGE_ID_EXTERNAL_REVIEW],
    'current_value' => 4,
    'updated_value' => 6,
],
```

here `state_id` is same that. So in first loop it will update value `2` to `4` . but in the third loop it will update that from `4` to `6` which is undesirable and we end up with unsynced value . One way I can think of updating it is to have a `updated_at` column and then only update those that has not updated in last `300` seconds . what you think ?

Reply:
Yes that could work. Another option is to do a 2-step process, with an intermediate value. Like: `2` -> `9994` -> `4`. But maybe there are better ways to handle such DB changes. Maybe ask in #pkp-dev to see if anyone has good ideas?

Reply:
fixed it .

Reply:
Looks good. I had a look at pkp-lib. Do you need to add the `updated_at` column to the `SubmissionsMigration` for fresh installs?

Reply:
> Do you need to add the `updated_at` column to the `SubmissionsMigration` for fresh installs?

no need for that . right now we do not have any `timestamp` functionality which used by laravel models . So it will be attached, then updated and then removed . no point in keeping something in DB that we will not use right now.

Reply:
Ah, right! I didn't see that it was removed during the upgrade. Good thinking!